syntax = "proto3";

service GameMaster {
  rpc AddProcess(AddProcessRequest) returns (AddProcessResponse) {}
  rpc CreateChain(CreateChainRequest) returns (CreateChainResponse) {}
  rpc ListChain(ListChainRequest) returns (ListChainResponse) {}
}

message AddProcessRequest {
  string name = 1;
}

message AddProcessResponse {}

message CreateChainRequest {}

message CreateChainResponse {
  repeated string chain = 1;
}

message ListChainRequest {}

message ListChainResponse {
  string chain = 1;
}
